zephyr/CMakeLists.txt

```diff
@@ -398,6 +398,7 @@ if (CONFIG_SOC_SERIES_NXP_IMX8)
 		${SOF_PLATFORM_PATH}/imx8/lib/dai.c
 		${SOF_PLATFORM_PATH}/imx8/lib/dma.c
 		${SOF_PLATFORM_PATH}/imx8/lib/memory.c
+		${SOF_SRC_PATH}/schedule/ll_schedule.c
```

Comment:
do I understand correctly that this way you include both ll_zephyr.c and ll_schedule.c in your builds? Do you need them both? Or would it be better to exclude ll_zephyr.c on imx for now?

Reply:
@lyakh my understanding is that we can skip using ll_zephyr.c for now for IMX.

Reply:
Yes, for now, for i.MX, the ll_zephyr.c and ll_schedule.c should be exclusive. This way we will not have the problem I've mentioned [here](https://github.com/thesofproject/sof/pull/4503#discussion_r673398819). - @marc-hb 
But I saw that ll_zephyr.c is mandatory included for all platform - see [here](https://github.com/thesofproject/sof/blob/main/zephyr/CMakeLists.txt#L482).
I can removed it from the mandatory files list (`[# Mandatory Files used on all platforms.]`(https://github.com/thesofproject/sof/blob/main/zephyr/CMakeLists.txt#L436) ) and add it for Intel's platforms.
@lyakh It will be ok this solution?

Reply:
yes, should be fine. And as you do that, I think it'd be better to also keep ll_schedule.c for `CONFIG_SOC_SERIES_INTEL_ADSP_BROADWELL` and  `CONFIG_SOC_SERIES_INTEL_ADSP_BAYTRAIL` even though they aren't really supported under Zephyr ATM.

Reply:
> yes, should be fine. 

Awesome, @iuliana-prodan I trust you will submit these CMake changes then so I can rebase and adjust #4503 on top of them? In #4525 you can see I tried really hard to compile IMX and I made some progress cherry-picking left and right but it's still failing to build for me.

Reply:
@lyakh @marc-hb I've made the CMake changes.
Please check if I've missed something for Intel's platforms.

Reply:
The CMake changes make sense to me but I don't understand why they're split over two different commits?